cmd/kubelet/app/options/options.go

```diff
@@ -69,6 +69,7 @@ func NewKubeletServer() *KubeletServer {
 		KubeletConfiguration: componentconfig.KubeletConfiguration{
 			Address:                     "0.0.0.0",
 			CAdvisorPort:                4194,
+			CalculateVolumeDiskUsageFrequency: unversioned.Duration{5 * time.Minute},
```

Comment:
Its just that 5 minutes is a long enough time to fill up the disk quite a
bit. Even if the threshold was set to say 80% for example, we might still
end up failing to detect this for 5 minutes.
How about starting with a minute and reducing if it proves to be expensive?

On Thu, Feb 4, 2016 at 8:47 AM, Phillip Wittrock <notifications@github.com>
wrote:

> In cmd/kubelet/app/options/options.go
> <https://github.com/kubernetes/kubernetes/pull/19741#discussion_r51900328>
> :
>
> > @@ -69,6 +69,7 @@ func NewKubeletServer() *KubeletServer {
> >  		KubeletConfiguration: componentconfig.KubeletConfiguration{
> >  			Address:                     "0.0.0.0",
> >  			CAdvisorPort:                4194,
> > +			CalculateVolumeDiskUsageFrequency: unversioned.Duration{5 * time.Minute},
>
> I am fine with reducing this, however the alerting disk thresholds should
> allow for more than 5 minutes of runway before the disk is full to give
> proper reaction time. I expect disk usage to be much less volatile than
> other metrics (cpu, memory, qps) and require less frequent polling. Since
> we don't have automated eviction, human intervention will be required to
> free up disk, so 1 second vs 1 minute of latency shouldn't be much
> different. Thoughts?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/19741/files#r51900328>.
>